### ISSUE TITLE
Empty space when only 2 apps are open

### ISSUE THREAD

**Issue by e391hed12** [ramensoftware/windows-11-taskbar-styling-guide#101]
![image](https://github.com/user-attachments/assets/44c91f54-6ba9-4fab-b616-a9513e82a92b)

Is there a way to eliminate that space to the right of the Opera icon?

**Comment by bbmaster123** [comment]
try this?

```
{
  "theme": "DockLike",
  "controlStyles[0].target": "Microsoft.UI.Xaml.Controls.ItemsRepeater#TaskbarFrameRepeater",
  "controlStyles[0].styles[0]": "HorizontalAlignment=1",
  "controlStyles[0].styles[1]": "Margin=-4,0,0,0"
}
```

might need to play a bit with that margin, -4px looks correct to me but I didn't measure it haha
cheers :)

**Comment by e391hed12** [comment]
Thanks a lot!!

**Comment by LeChocolatChaud** [comment]
After some investigations with UWPSpy, I found this element set to a fixed width:

`Taskbar.Gripper#GripperControl`

![Taskbar.Gripper#GripperControl](https://github.com/user-attachments/assets/d89b4301-e58b-4333-9e23-f118eb42db9e)

Setting its width to `Auto` completely eliminated the empty space.

![Result effect](https://github.com/user-attachments/assets/f13cc054-f841-4634-805c-ea44dd1066dd)

**Comment by bbmaster123** [comment]
very interesting, it does work, and also, setting its width to a static value seems to set a minimum width for the dock, while still letting it expand:
![image](https://github.com/user-attachments/assets/74e3c69b-9aef-4594-a811-6e4d82f6b249)
![image](https://github.com/user-attachments/assets/abafcadf-ff6b-415d-957c-8be20795b268)
Thanks for sharing!!

**Comment by m417z** [comment]
@bbmaster123 @AmberWat how about setting width to `Auto` in the theme? Any downsides?

**Comment by AmberWat** [comment]
I'm not sure what the gripper does, if anything. It looks like it's meant to be some sort of handle for dragging but it seems to be left unused.
If we wanted to be super safe we could probably do `Width=Auto` and combine it with something like `MinWidth=24` so that it's still there *if* something needs it.

**Comment by m417z** [comment]
I suspect it might be the tablet mode thing. Should be possible to verify with [the tablet posture registry option](https://www.elevenforum.com/t/turn-on-or-off-tablet-optimized-taskbar-in-windows-11.5133/).

![image](https://github.com/user-attachments/assets/a56c8dac-df02-4c07-a273-3ded550ecf1b)

**Comment by bbmaster123** [comment]
@m417z 

Yes it looks like the gripper is specifically the grey horizontal line in the tablet mode taskbar:
![image](https://github.com/user-attachments/assets/8ef3438a-e25f-4d3f-9542-c2cf9806cff0)

shouldn't be a problem as long as we include MinWidth. Without it, the horizontal grey line is missing. Other than that, it makes it a bit hard to right click the taskbar. We could also add a 5px margin to the framerepeater to mitigate this:

```
{
  "theme": "",
  "controlStyles[0].target": "Taskbar.Gripper#GripperControl",
  "controlStyles[0].styles[0]": "Width=Auto",
  "controlStyles[0].styles[1]": "MinWidth=24",
  "controlStyles[1].target": "Microsoft.UI.Xaml.Controls.ItemsRepeater#TaskbarFrameRepeater",
  "controlStyles[1].styles[0]": "Margin=5,0,5,0"
}
```
Otherwise, we could change the BackgroundFill in the theme from Collapsed to Opacity=0

**Comment by m417z** [comment]
> Otherwise, we could change the BackgroundFill in the theme from Collapsed to Opacity=0

I think that's fine, the taskbar is not clickthrough anyway (https://github.com/ramensoftware/windhawk-mods/issues/805), at least the empty space could be used for the menu and perhaps [click actions with the mod](https://windhawk.net/mods/taskbar-empty-space-clicks).

**Comment by bbmaster123** [comment]
Fair enough. 
Thoughts on 5px on each side vs 0px or some other value?
I can push it through tonight

**Comment by AmberWat** [comment]
> @m417z
> 
> Otherwise, we could change the BackgroundFill in the theme from Collapsed to Opacity=0

That doesn't work, at least when I test it. `RootGrid` is the element that controls where right clicking summons the menu.

> Other than that, it makes it a bit hard to right click the taskbar. We could also add a 5px margin to the framerepeater to mitigate this

Well, I actually did put a slight padding on `RootGrid`. It doesn't make sense to have properties on multiple elements working towards the same thing.
Maybe get rid of the margin on `TaskbarFrameRepeater` and instead up the padding on `Taskbar.TaskbarFrame#TaskbarFrame > Grid#RootGrid` to `Padding=6,0,6,0`? (I think a total padding of 6 looks best)

**Comment by bbmaster123** [comment]
@AmberWat 
done!
feel free to ping me if any more changes are needed :)

**Comment by ksb2311** [comment]
![image](https://github.com/user-attachments/assets/934a2ac7-9b72-4a36-a338-47d15597d701)
Is there any way to separate widget to left from dock in DockLike

**Comment by alxkum** [comment]
I noticed that with the Docklike theme there's no transition animations whenever a new element is added or one is removed (e.g. a window is closed). The other themes all seem to have the animation, which makes it much easier to track where a new window was just opened in the taskbar or which one I just closed.  

Is there any way I can adjust this with the settings?
